clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1634,7 +1635,8 @@ public Map<TopicPartition, Long> beginningOffsets(Collection<TopicPartition> par
      * @param partitions the partitions to get the earliest offsets
      * @param timeout The maximum amount of time to await retrieval of the beginning offsets
      *
-     * @return The earliest available offsets for the given partitions
+     * @return The earliest available offsets for the given partitions. If the offset for a specific partition cannot be 
```

Comment:
the returned value is inconsistent between classic consumer and async consumer. I have opened https://issues.apache.org/jira/browse/KAFKA-19205 to trace it. this PR should be pending for the ticket